src/rules/relay_connections_have_fields.js

```diff
@@ -0,0 +1,38 @@
+import { ValidationError } from '../validation_error';
+const MANDATORY_FIELDS = ['pageInfo', 'edges'];
+
+function validateConnection(context, node) {
+  const typeName = node.name.value;
+  if (!typeName.endsWith('Connection')) {
+    return;
+  }
+  const fieldNames = node.fields.map(field => field.name.value);
+  const missingFields = MANDATORY_FIELDS.filter(
+    requiredField => fieldNames.indexOf(requiredField) === -1
+  );
+
+  if (missingFields.length) {
+    context.reportError(
+      new ValidationError(
+        'relay-connections-have-fields',
+        `Connection \`${typeName}\` is missing the following field${missingFields.length >
+        1
+          ? 's'
+          : ''}: ${missingFields.join(', ')}.`,
+        [node]
+      )
+    );
+  }
+}
+
+export function RelayConnectionsHaveFields(context) {
+  return {
+    ObjectTypeDefinition(node) {
+      validateConnection(context, node);
+    },
+
+    InterfaceTypeDefinition(node) {
```

Comment:
I might be reading the spec wrong, but it appears like interfaces cannot end in "Connection":

> Any type whose name ends in “Connection” is considered by Relay to be a Connection Type. Connection types must be an “Object” as defined in the “Type System” section of the GraphQL Specification.

https://facebook.github.io/relay/graphql/connections.htm#sec-Connection-Types

Under the "Type System" section of GraphQL specification, "Object" appears to mean object types, not interface types: http://facebook.github.io/graphql/draft/#sec-Objects

Reply:
Indeed.